### PR TITLE
Add jest to modules

### PR DESCRIPTION
**Pull request by MarcusTXK**
# Description

Added jest to modules
Added pre-push hook for husky
Added tests for Repeat bundle

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Run `yarn test` to test jest
`git push` to test pre-push husky script

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,27 +4,41 @@
   "version": "1.0.0",
   "repository": "https://github.com/source-academy/modules.git",
   "license": "Apache-2.0",
+  "scripts-info": {
+    "module": "Add a new bundle or tab",
+    "lint": "Check code for errors",
+    "lint:fix": "Automatically fix lint errors",
+    "build": "Transpile modules' files from src folder to build folder",
+    "serve": "Start the http-server to serve .js files in build folder",
```
Comment (Anonymxtrix) [inline package.json:12]: Not just `.js` files, its all files from build folder (and also the same directory structure).

File: package.json [hunk 2]
```diff
@@ -58,5 +74,11 @@
   },
   "peerDependencies": {
     "tslib": "^2.1.0"
+  },
+  "jest": {
+    "roots": [
+      "src/bundles",
+      "src/tabs"
+    ]
```
Comment (Anonymxtrix) [inline package.json:82]: Wanna add testing capability to our generate template app as well? Might be a good idea idk.

**Review by Anonymxtrix — COMMENTED** [review]
Solid work haha! 👍👍👍👍👍

File: package.json (hunk 1 above)
Comment (MarcusTXK) [reply package.json:12]: Good catch, will fix that

**Comment by MarcusTXK** [comment]
Thanks for the feedback! Made the necessary edits

**Review by Anonymxtrix — APPROVED** [review]

**Pull request merged**